lib/kubernetes-deploy/restart_task.rb

```diff
@@ -43,14 +44,18 @@ def perform(deployments_names = nil)
       resources = build_watchables(deployments, start)
       ResourceWatcher.new(resources, logger: @logger, operation_name: "restart").run
       success = resources.all?(&:deploy_succeeded?)
+      error = :timeout if !success && resources.any?(&:deploy_timed_out?)
+      [success, error]
     rescue FatalDeploymentError => error
       @logger.summary.add_action(error.message)
       success = false
+      [success, error]
     ensure
       @logger.print_summary(success)
       status = success ? "success" : "failed"
       tags = %W(namespace:#{@namespace} context:#{@context} status:#{status} deployments:#{deployments.to_a.length}})
       ::StatsD.measure('restart.duration', StatsD.duration(start), tags: tags)
+      [success, error]
```

Comment:
As you mentioned in your comment above `ensure`s don't return, so this can be deleted, right?